Kernel/Term.cpp

```diff
@@ -37,7 +37,9 @@ constexpr unsigned Term::SPECIAL_FUNCTOR_LOWER_BOUND;
 
 void Term::setId(unsigned id)
 {
-  if (env.options->randomTraversals()) {
+  if (env.options->randomTraversals() &&
+      Random::seed() != 1) { // not until a proper seed has been set (i.e. after parsing!)
```

Comment:
This seems wrong - if we don't set a random seed or set it to 1, we won't randomise `id`. Perhaps we could abuse `env.statistics->phase` here?

Reply:
I think I tested it. The point it we now only set Options::randomSeed -> Random::seed() after parsing. And randomizing termIds during parsing is evil, because then it's hard to sync up the portfolio master->worker setting with the vampire mode (plain single-strategy) setting. Setting random seed to 1 and wanting randomized termId's is something I can live without. (A random random seed is almost never equal to 1.) On the other hand, relying on phase means I will need to make sure nobody messes around with phase. And it's dealt with in so many places that I don't trust it atm.

Reply:
OK. Might try and set some kind of unreasonable default value at some point so we can detect it, but low priority. Thanks for the explanation!